simulation/traffic_simulator/src/math/hermite_curve.cpp

```diff
@@ -163,55 +163,48 @@ boost::optional<double> HermiteCurve::getCollisionPointIn2D(
   return *std::min_element(s_values.begin(), s_values.end());
 }
 
+boost::optional<double> HermiteCurve::solveNewtonMethod(
+  const geometry_msgs::msg::Point & point, double initial_value, double tolerance,
+  unsigned int max_iteration) const
+{
+  double ret = initial_value;
+  for (unsigned i = 0; i < max_iteration; i++) {
+    double error = getSquaredDistanceIn2D(point, ret);
+    if (std::fabs(error) < (tolerance * tolerance)) {
+      return ret;
+    }
+    ret = ret - getNewtonMethodStepSize(point, ret);
+    if (ret < 0 || ret > 1) {
+      return boost::none;
+    }
+  }
+  return ret;
+}
+
 boost::optional<double> HermiteCurve::getSValue(
-  geometry_msgs::msg::Point point, double threshold_distance, unsigned int initial_resolution,
+  const geometry_msgs::msg::Point & point, unsigned int initial_resolution,
   unsigned int max_iteration, double tolerance, bool autoscale) const
```

Comment:
`unsigned int` => `std::size_t`

https://stackoverflow.com/questions/131803/unsigned-int-vs-size-t

Reply:
fixed in https://github.com/tier4/scenario_simulator_v2/pull/603/commits/0b3461ec444280b0c20ed2b8e5b008d42e74cc5e